modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/ColocationGroup.java

```diff
@@ -92,8 +92,8 @@ public ColocationGroup local(UUID nodeId) {
                     .collect(Collectors.toList());
         }
 
-        return new ColocationGroup(Arrays.copyOf(sourceIds, sourceIds.length), Collections.singletonList(nodeId),
-                locAssignments);
+        return new ColocationGroup(sourceIds == null ? null : Arrays.copyOf(sourceIds, sourceIds.length),
```

Comment:
Does any test cover this? Or It just fails somewhere else?

Reply:
LocalQueryIntegrationTest fails without this change